src/main/webapp/WEB-INF/view/admin.jsp

```diff
@@ -14,11 +14,42 @@
    See the License for the specific language governing permissions and
    limitations under the License.
  --%>
+<%@ page import= "codeu.model.data.Conversation"%>
+<%@ page import= "codeu.model.data.Message"%>
+<%@ page import= "codeu.model.data.User"%>
+<%@ page import= "codeu.model.store.persistence.PersistentDataStoreException"%>
+<%@ page import= "codeu.model.store.persistence.PersistentDataStore"%>
+<%@ page import= "com.google.appengine.api.datastore.DatastoreService"%>
+<%@ page import= "com.google.appengine.api.datastore.DatastoreServiceFactory"%>
+<%@ page import= "com.google.appengine.api.datastore.Entity"%>
+<%@ page import= "com.google.appengine.api.datastore.PreparedQuery"%>
+<%@ page import= "com.google.appengine.api.datastore.Query"%>
+<%@ page import= "com.google.appengine.api.datastore.Query.SortDirection"%>
+<%@ page import="codeu.model.store.basic.UserStore" %>
+<%@ page import= "java.time.Instant"%>
+<%@ page import= "java.util.ArrayList"%>
+<%@ page import= "java.util.List"%>
+<%@ page import= "java.util.UUID"%>
 <t:base>
     <jsp:attribute name="bodyContent">
+    <%
+       PersistentDataStore store = new PersistentDataStore();
+
+       int numusers = store.loadUsers().size();
+
+       int numconvo = store.loadConversations().size();
+
+       int nummessage = store.loadMessages().size();
```

Comment:
It seems that this kind of logic should live in the controller, not the view; evidenced by the huge amount of imports (although I think if you just import the PersistentDataStore you should be fine). Also, I dont think we should be accessing the PersistentDataStore directly; so far we have been abstracting that away through the stores of each Model